synedrion/src/tools/secret.rs

```diff
@@ -39,22 +73,349 @@ impl<T: Zeroize> Debug for Secret<T> {
     }
 }
 
-impl<T: Zeroize> From<SecretBox<T>> for Secret<T> {
-    fn from(value: SecretBox<T>) -> Self {
-        Self(value)
+impl<T> Secret<T>
+where
+    T: Zeroize + Clone + HasWide,
+    T::Wide: Zeroize,
+{
+    pub fn to_wide(&self) -> Secret<<T as HasWide>::Wide> {
+        Secret::init_with(|| {
+            let mut value = self.expose_secret().clone();
+            let result = value.to_wide();
+            value.zeroize();
+            result
+        })
+    }
+
+    pub fn mul_wide(&self, rhs: &T) -> Secret<T::Wide> {
+        Secret::init_with(|| {
+            let mut value = self.expose_secret().clone();
+            let result = value.mul_wide(rhs);
+            value.zeroize();
+            result
+        })
+    }
+}
+
+impl<T> Secret<Signed<T>>
+where
+    T: Zeroize + Clone + Encoding + Integer + HasWide + ConditionallySelectable + crypto_bigint::Bounded,
+    T::Wide: ConditionallySelectable + crypto_bigint::Bounded + Zeroize,
+{
+    pub fn to_wide(&self) -> Secret<Signed<<T as HasWide>::Wide>> {
+        Secret::init_with(|| {
+            let mut value = *self.expose_secret();
+            let result = value.to_wide();
+            value.zeroize();
+            result
+        })
```

Comment:
For my education, can you explain why doing what you're doing here is better than:

```rust
Secret::init_with(||
	self.expose_secret().to_wide()
)
```

Reply:
Hm, good catch. I think it is some remainder from the previous iterations, but I don't think it makes any sense now.

Reply:
Fixed